pkg/apis/flowcontrol/validation/validation.go

```diff
@@ -385,8 +385,8 @@ func ValidatePriorityLevelConfigurationSpec(spec *flowcontrol.PriorityLevelConfi
 // ValidateLimitedPriorityLevelConfiguration validates the configuration for an execution-limited priority level
 func ValidateLimitedPriorityLevelConfiguration(lplc *flowcontrol.LimitedPriorityLevelConfiguration, fldPath *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
-	if lplc.AssuredConcurrencyShares <= 0 {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("assuredConcurrencyShares"), lplc.AssuredConcurrencyShares, "must be positive"))
+	if lplc.NominalConcurrencyShares <= 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("nominalConcurrencyShares"), lplc.NominalConcurrencyShares, "must be positive"))
```

Comment:
because the field name differs by version, you'll need to pass in the request version and use the appropriate field name based on the request version.

See https://github.com/kubernetes/kubernetes/pull/111506/files as an example

Reply:
Thanks for the pointer, I was going to ask about that.

Reply:
Just curious as to why we do typed validation on the internal object, could we not validate on the typed request object? (I am not proposing any change, just want to understand the reason or historical context behind it)

Reply:
so that validation is consistent no matter what version the request came in via (or where there are differences, they are intentional and clearly visible). As complex as some of our validation is, requiring replicating it against multiple external versions would surely drift and introduce validation gaps via some versions

Reply:
so we traded off additional complexity in the validation logic to avoid the drift, thanks for the explanation!